### PR TITLE
DBZ-2623 Build checkstyle rules in GH Action

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2623

### CODE REVIEW

File: .github/workflows/formatting-check.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
           ${{ runner.os }}-maven-
 
     - name: Build configuration
-      run: mvn clean install -pl :debezium-ide-configs
+      run: mvn clean install -pl :debezium-ide-configs,:debezium-checkstyle
```
Comment (gunnarmorling) [inline .github/workflows/formatting-check.yml:24]: So thinking about it, I think this is what should be run here:

    mvn clean package -DskipTests=true -DskipITs=true process-sources -Dformat.formatter.goal=validate -Dformat.imports.goal=check

Comment (gunnarmorling) [reply .github/workflows/formatting-check.yml:24]: This also means Checkstyle, Impsort and Format check can be skipped in the main test run which save some time there.

File: debezium-core/src/main/java/io/debezium/config/Configuration.java [hunk 2]
```diff
@@ -59,7 +59,9 @@
  */
 @Immutable
 public interface Configuration {
-
+;
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/config/Configuration.java:62]: Left-over?

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/config/Configuration.java]: No, test :-)

File: .travis.yml [hunk 3]
```diff
@@ -4,18 +4,18 @@ cache:
...
-  - MAVEN_CLI: '"clean install -B -pl debezium-connector-mongodb -am -Passembly -Dformat.formatter.goal=validate -Dformat.imports.goal=check -Dversion.mongo.server=4.2 -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"'
-  - MAVEN_CLI: '"clean install -B -pl :debezium-quarkus-outbox -am -amd -Passembly -Dformat.formatter.goal=validate -Dformat.imports.goal=check -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"'
-  - MAVEN_CLI: '"clean install -B -pl debezium-server -am -amd -Dformat.formatter.goal=validate -Dformat.imports.goal=check -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"'
-  - MAVEN_CLI: '"clean install -B -pl debezium-testing -am -amd -Passembly -Dformat.formatter.goal=validate -Dformat.imports.goal=check -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"'
+  - MAVEN_CLI: '"clean install -B -pl debezium-connector-sqlserver -am -Passembly -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"'
```
Comment (gunnarmorling) [inline .travis.yml:7]: So can we actually skip format and impsort check? At least `-Dcheckstyle.skip=true` should be doable.

**Comment by jpechane** [comment]
@gunnarmorling Formatting, impsort and checktyle are disabled for test build/execution.  Moreover API compatibility check should be run also only as the part of the formatting check action.

**Comment by gunnarmorling** [comment]
Nice, thx!

**Pull request merged**